src/url.js

```diff
@@ -407,42 +422,22 @@ export function isAmpScriptUri(uri) {
  * @return {?string}
  */
 export function getProxyServingType(url) {
-  if (typeof url == 'string') {
-    url = parseUrlDeprecated(url);
-  }
+  url = urlAsLocation(url);
   if (!isProxyOrigin(url)) {
     return null;
   }
   const path = url.pathname.split('/', 2);
   return path[1];
 }
 
-/**
- * Returns whether the URL origin is localhost.
- * @param {string|!Location} url URL of an AMP document.
- * @return {boolean}
- */
-export function isLocalhostOrigin(url) {
-  if (typeof url == 'string') {
-    url = parseUrlDeprecated(url);
-  }
-  return urls.localhostRegex.test(url.origin);
-}
-
 /**
  * Returns whether the URL has valid protocol.
  * Deep link protocol is valid, but not javascript etc.
  * @param {string|!Location} url
  * @return {boolean}
  */
 export function isProtocolValid(url) {
-  if (!url) {
-    return true;
-  }
-  if (typeof url == 'string') {
-    url = parseUrlDeprecated(url);
-  }
-  return !INVALID_PROTOCOLS.includes(url.protocol);
+  return !url || !INVALID_PROTOCOLS.includes(urlAsLocation(url).protocol);
```

Comment:
Nit: I prefer to think in positives instead of negatives.

```suggestion
  return !!url && !INVALID_PROTOCOLS.includes(urlAsLocation(url).protocol);
```

Reply:
This change as suggested is fails tests, but I turned it into `!(url && INVALID_PROTOCOLS.includes(urlAsLocation(url).protocol))`